src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt

```diff
@@ -13,7 +20,33 @@ class MatrixRoomMessageReceivedListener(private val plugin : Main) : RoomMessage
             return true
         }
 
+        if (event.msg.content.msgtype == MessageTypes.IMAGE) {
+            val messageContent = event.msg.content as ImageMessageContent
+            val imageType = messageContent.mimetype.split('/').last().toUpperCase()
+            val url = getMediaUrl(plugin, messageContent)
+            sendLinkMessage(plugin, url, "Image [${messageContent.width}x${messageContent.height}, $imageType]")
+            return true
+        }
+
+        if (event.msg.content.msgtype == MessageTypes.FILE) {
+            val messageContent = event.msg.content as FileMessageContent
+            val url = getMediaUrl(plugin, messageContent)
+            val size = FileUtils.byteCountToDisplaySize(messageContent.size!!.toLong())
+            sendLinkMessage(plugin, url, "File [${messageContent.filename}, $size]")
+            return true
+        }
+
         plugin.server.broadcastMessage("[§2${event.msg.sender}§r] ${event.msg.body}")
         return true
     }
+
+    private fun sendLinkMessage(plugin: Main, url: String, label: String) {
```

Comment:
There is no need for passing `plugin` to this method -- it's already available as an instance variable.  Please remove it from the both method signatures.

Reply:
Oida